zipline/data/hdf5_daily_bars.py

```diff
@@ -132,6 +146,10 @@
 END_DATE = 'end_date'
 
 
+# XXX is reserved for "transactions involving no currency".
+MISSING_CURRENCY = 'XXX'
```

Comment:
Is there a distinction here between these "transactions involving no currency" and what we have been using for the missing currency value (`b'\0\0\0'`), or should we maybe unify those?

Reply:
Hmm. That's a good question. My reason for using `'XXX'` here is that it felt like a slightly more "user-friendly" missing value than `b'\0\0\0'` (which numpy displays as `b''`, which can often create confusion for users). But I agree it seems not great to have multiple "missing"/"unknown" representations for currency data.

Reply:
I punted on this for the time being, but we should circle back on this with @jmccorriston before we finalize this feature.

Reply:
What is the context in which this would be visible to a user?

Reply:
Currently, this value wouldn't be user-visible, but there are other contexts where we need a notion of a "missing" currency which might be user-visible, and it'd be nice to be consistent.

I think where we landed on this for discussion the other day is that any user-facing currency columns should have a missing value of `None` (like other pipeline string columns). `None` isn't an option in this particular case (we're storing arrays of fixed-length strings, which can't store python objects like `None`), so no matter what we're going to have to convert from an internal representation to a user-facing representation, which means we've got a good amount of room to be flexible here.